pkg/argocd/argocd.go

```diff
@@ -37,19 +37,31 @@ func Apply(ctx context.Context, config *rest.Config, namespace, argoImage string
 	if err != nil {
 		return err
 	}
+
 	deployments, err := clientset.AppsV1().Deployments(namespace).List(metav1.ListOptions{
 		LabelSelector: "app.kubernetes.io/part-of=argocd",
 	})
 	if err != nil {
 		return err
 	}
-	expectedCount := 4
-	foundCount := len(deployments.Items)
-	if foundCount == expectedCount {
-		klog.Infof("Found %d of %d deployments", foundCount, expectedCount)
+	expectedDeploymentCount := 3
+	foundDeploymentCount := len(deployments.Items)
+
+	statefulsets, err := clientset.AppsV1().StatefulSets(namespace).List(metav1.ListOptions{
+		LabelSelector: "app.kubernetes.io/part-of=argocd",
+	})
+	if err != nil {
+		return err
+	}
+	expectedStatefulSetCount := 1
+	foundStatefulSetCount := len(statefulsets.Items)
+
+	if foundDeploymentCount == expectedDeploymentCount && foundStatefulSetCount == expectedStatefulSetCount {
+		klog.Infof("Found %d of expected %d deployments, found %d of expected %d statefulsets, skip", foundDeploymentCount, expectedDeploymentCount, foundStatefulSetCount, expectedStatefulSetCount)
```

Comment:
@srueg Do you see a benefit for logging this always or can we just remove the log message? Just it fills up the Kibana.

Reply:
Good point. Changed the logic.